lib/gameboard/board.rb

```diff
@@ -0,0 +1,517 @@
+    #                 0 | 1 | 2 | 3 |
+    #                   -------------
+    #                 1 | A | B | C |
+    #                   -------------
+    #                 2 | X | Y | Z |
+    #                   -------------
+    #                   - 0 - 1 - 2 -
+    #
+    #   neighbors([1,1])
+    #           #=> [1, 2, 3, "A", "C", "X", "A", 1]
+    #
+    #   neighbors([1,1] true)
+    #           #=> [
+    #                [0, 2], [1, 2],
+    #                [2, 2], [0, 1],
+    #                [2, 1], [0, 0],
+    #                [0, 1], [0, 2]
+    #               ]
+    #
+    #
+    def neighbors(point, coords = false)
+      temp = Coordinate.new(point[0], point[1])
+      valid_neighbors(temp).map do |cell|
+        coords ? cell.coord.position : cell.value
+      end
+    end
+
+    # Public: Set random cells to a given value.
+    #
+    # piece - A value to set to cells
+    #
+    def randomize(piece)
+      rand(1...board.length).times do
+        board[rand(board.length)].value = piece
+      end
+    end
+    # Public: Set a cells value at a given coordinate
+    #
+    # coord - A coordinate pair in the form X, Y
+    def set_cell(coord, value)
+      cell = find_cell(coord)
+      raise "Cell Not Found" unless !!cell
+      cell.value = value
+    end
+
+    # Public: Return every column on the gameboard.
+    #
+    # coords: An optional boolean to specify returning an Array of coordinates
+    #         instead of values.
+    #
+    # Example
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #
+    #   vertical
+    #     #=> [
+    #          ["X", "A", 1],
+    #          ["Y", "B", 2],
+    #          ["Z", "C", 3]
+    #         ]
+    #
+    #   vertical(true)
+    #     #=> [
+    #          [[0, 0], [0, 1], [0, 2]],
+    #          [[1, 0], [1, 1], [1, 2]],
+    #          [[2, 0], [2, 1], [2, 2]]]
+    #         ]
+    #
+    def vertical(coords = false)
+      columns = []
+      width.times do |x|
+        columns << board.select { |cell| cell.coord.x == x }.map do |cell|
+          coords ? cell.coord.position : cell.value
+        end
+      end
+      columns
+    end
+
+    private
+      # Internal: Returns the gameboard array.
+      attr_reader :board
+      # Internal: Returns the default gameboard cell value
+      attr_reader :cells
+      # Internal: Stores whether the board is nailed down
+      attr_reader :nailed
+
+      # Internal: return a diagonal array given a starting point and a
+      #           slope.
+      def get_diagonal(start, coords, slope = true)
+        oper = (slope == true ? :+ : :-)
+        diagonal = (0...height).map do |i|
+          position = [start[0] + i, start[1].send(oper, i)]
+          if ((0...width).include?(start[0] + i) && (0...height).include?(start[1].send(oper, i)))
```

Comment:
It's a bit crammed in here, though not completely incomprehensible.
However, since you're just compacting out the nils anyhow, do you even need the conditional?
If you're doing it for performance reasons, I think there could be some additional methods on the Board / Coordinate class to deal with this logic. Coordinate.new(0,4).delta(0,1) could return a new coordinate derived from the original & given delta. This new Coordinate could then be passed into a Board.valid_coordinate?